### PR TITLE
Add SSD example

### PR DESCRIPTION
**Pull request by slin07**
This PR:
(1) adds an mobile-ssd conversion example (only feature extractor involved)
(2) fix bugs involving reshape->squeeze axis alignment

### CODE REVIEW

**Comment by FrozenGene** [comment]
@slin07 @aseemw Could we explain the mechanism of Reshape->Squeeze here? I find that in SSD-mobilenet example has Reshape->Squeeze. For example, (24, 1, 1) ---> Reshape([1, 6, 1, 4] [N, H,  W, C] , here we are the same as Tensorflow. But we skip the op of Squeeze and "hack" Reshape->Squeeze.

```python
  # When reshape is immediately followed by squeeze
  if len(op.outputs) > 0 and len(op.outputs[0].consumers()) > 0 and \
      op.outputs[0].consumers()[0].type == 'Squeeze':
    squeezed_output_name = compat.as_str_any(
        op.outputs[0].consumers()[0].outputs[0].name)
    target_shape = context.shape_dict[squeezed_output_name]
``` 
After this step, target shape is changed as  [1, 6, 4] [H W C] and this will be passed:
```python
  elif len(target_shape) == 3:
    # Target shape is [H,W,C] --> [1, C, H, W]
    new_shape = (1, target_shape[2], target_shape[0], target_shape[1])
    context.builder.add_reshape(
        output_name, input_name, output_name, new_shape, 1)
```

We have the target shape [1, 4, 1, 6] [ N, C, H, W]. We can see that previously is [1, 6, 1, 4] [N, H,  W, C], Now, [1, 4, 1, 6] [ N, C, H, W] ---> [1, 1, 6, 4] [N H W C]. That is H W is swapped. 

But if I run the SSD-mobilenet, Nerual Network compiler shows that: 

> Neural Network compiler 212: 300 , name = BoxPredictor_5/Reshape:0, output shape : (C,H,W) = (4, 1, 6)

But its output is : 
[array([[[  6.16023317e-03],
        [ -6.97045308e-03],
        [  1.58075150e-02],
        [ -7.08721653e-02],
        [ -7.34762289e-03],
        [  8.28582793e-04]],

       [[ -3.00084054e-01],
        [ -1.56376868e-01],
        [ -3.97226036e-01],
        [ -7.84830600e-02],
        [ -1.41842425e-01],
        [ -3.00244123e-01]],

       [[  1.66258097e-01],
        [  1.82900560e+00],
        [ -1.53803492e+00],
        [  5.30515552e-01],
        [ -2.34279752e+00],
        [  3.40440273e-02]],

       [[ -1.12962812e-01],
        [ -1.64404511e+00],
        [  1.64561009e+00],
        [ -2.36857176e+00],
        [  7.34788895e-01],
        [ -2.53201723e-01]]])]

The shape is (4, 6, 1) [C H W]. Not (4, 1 6) [C H W] showed by Neural Network Compiler's output

Could you help to explain? I am spending much time here and can not go to further place...
